modules/sources/contracts/fetch-pool-type-data.ts

```diff
@@ -0,0 +1,44 @@
+import { ViemClient } from '../types';
+import { stableContractCalls, parseStableContractCalls } from './pool-type-data';
+import { multicallViem, ViemMulticallCall } from '../../web3/multicaller-viem';
+import { PrismaPoolType } from '@prisma/client';
+
+export const fetchPoolTypeData = async (
+    client: ViemClient,
+    pools: {
+        id: string;
+        type: PrismaPoolType;
+    }[],
+    blockNumber?: bigint,
+) => {
+    const calls = pools
+        .flatMap((pool) => {
+            switch (pool.type) {
+                case PrismaPoolType.STABLE:
+                    return stableContractCalls([pool.id]);
+            }
+        })
+        .filter((x): x is ViemMulticallCall => !!x);
+
+    const results = await multicallViem(client, calls, blockNumber);
+
+    const params = pools
+        .map(({ id, type }) => {
+            const result = results[id];
+
+            if (result === undefined) {
+                return;
+            }
+
+            switch (type) {
+                case PrismaPoolType.STABLE:
+                    return {
+                        id,
+                        typeData: parseStableContractCalls(result),
+                    };
+            }
+        })
+        .filter((x): x is { id: string; typeData: { amp: string } } => !!x);
```

Comment:
Cant we just filter the undefined results?

Reply:
yes, that's what it's doing. filtering out faulty responses. do you mean some other way to do that?